spec/message_spec.rb

```diff
@@ -100,23 +107,23 @@ def days(num)
   end
 
   it "Throws an error if the message is missing the signature" do
-    expect { @message.validate("") }.to raise_exception(Siwe::InvalidSignature, "Signature doesn't match message.")
+    expect { @message.verify("", nil, nil, nil) }.to raise_exception(Siwe::InvalidSignature, "Signature doesn't match message.")
   end
 
   it "Throws an error if the message is not yet valid" do
     key = Eth::Key.new
     @message.address = key.address.to_s
     @message.not_before = (Time.now.utc + days(1)).iso8601
     signature = key.personal_sign(@message.prepare_message)
-    expect { @message.validate(signature) }.to raise_exception(Siwe::NotValidMessage, "Message not yet valid.")
+    expect { @message.verify(signature, nil, nil, nil) }.to raise_exception(Siwe::NotValidMessage, "Message not yet valid.")
   end
 
   it "Throws an error if the message address is not EIP-55 complient" do
     key = Eth::Key.new
     @message.address = key.address.to_s.downcase
     signature = key.personal_sign(@message.prepare_message)
     expect do
-      @message.validate(signature)
+      @message.verify(signature, nil, nil, nil)
     end.to raise_exception(Siwe::InvalidAddress,
                            "Address does not conform to EIP-55 or is invalid.")
```

Comment:
Since EIP-55 is not strictly enforced, an all-downcase address does in fact comply with EIP-55. However, it's up to you to choose only to accept checksummed addresses.